src/wp-admin/includes/image-edit.php

```diff
@@ -1026,12 +1038,177 @@ function wp_save_image( $post_id ) {
 		$meta['sizes'] = array_merge( $meta['sizes'], $img->multi_resize( $_sizes ) );
 	}
 
+	$resize_sizes = array();
+	foreach ( $old_metadata['sizes'] as $size_name => $size_details ) {
+		// If the target is 'nothumb', skip generating the 'thumbnail' size.
+		if ( 'nothumb' === $target && 'thumbnail' === $size_name ) {
+			continue;
+		}
+
+		if (
+				isset( $meta['sizes'][ $size_name ] ) && ! empty( $meta['sizes'][ $size_name ] ) &&
+				$meta['sizes'][ $size_name ]['file'] !== $old_metadata['sizes'][ $size_name ]['file']
+		) {
```

Comment:
Can we reduce the indentation here?

```suggestion
		if (
			isset( $meta['sizes'][ $size_name ] ) && ! empty( $meta['sizes'][ $size_name ] ) &&
			$meta['sizes'][ $size_name ]['file'] !== $old_metadata['sizes'][ $size_name ]['file']
		) {
```